tests/sampling/test_forward.py

```diff
@@ -540,6 +540,50 @@ def test_normal_scalar_idata(self):
             ppc = pm.sample_posterior_predictive(idata, return_inferencedata=False)
             assert ppc["a"].shape == (nchains, ndraws)
 
+    def test_external_trace(self):
+        nchains = 2
+        ndraws = 500
+        with pm.Model() as model:
+            mu = pm.Normal("mu", 0.0, 1.0)
+            a = pm.Normal("a", mu=mu, sigma=1, observed=0.0)
+            trace = pm.sample(
+                draws=ndraws,
+                chains=nchains,
+            )
+
+        # test that trace is used in ppc
+        with pm.Model() as model_ppc:
+            mu = pm.Normal("mu", 0.0, 1.0)
+            a = pm.Normal("a", mu=mu, sigma=1)
+
+        ppc = pm.sample_posterior_predictive(
+            trace=trace, model=model_ppc, return_inferencedata=False
+        )
+        assert list(ppc.keys()) == ["a"]
+
+    def test_external_trace_det(self):
+        nchains = 2
+        ndraws = 500
```

Comment:
We don't care about the actual draws

```suggestion
        tune=50
        nchains = 1
        ndraws = 50
```